packages/core/src/fhirmapper/transform.ts

```diff
@@ -560,6 +593,57 @@ function evalAppend(ctx: TransformContext, target: StructureMapGroupRuleTarget):
   return [{ type: 'string', value: (arg1 ?? '').toString() + (arg2 ?? '').toString() }];
 }
 
+/**
+ * Evaluates the "cast" transform.
+ *
+ * "Cast source from one type to another. target type can be left as implicit if there is one and only one target type known."
+ *
+ * See: https://build.fhir.org/mapping-language.html#7.8.0.8.2
+ *
+ * @param ctx - The transform context.
+ * @param target - The FHIR Mapping target definition.
+ * @returns The evaluated target values.
+ * @internal
+ */
+function evalCast(ctx: TransformContext, target: StructureMapGroupRuleTarget): TypedValue[] {
+  const arg1 = resolveParameter(ctx, target.parameter?.[0])?.[0];
+  const arg2 = resolveParameter(ctx, target.parameter?.[1])?.[0]?.value;
+  if (arg2 === 'string') {
+    return [{ type: 'string', value: arg1?.value?.toString() }];
+  }
+  return [arg1];
```

Comment:
Should this be more defensive?  I'm not too familiar with how this is used in practice

Reply:
In short term for the C-CDA maps, this covers 100% of our needs.

In longer term, yeah, there's a lot of work here.  The FHIRPath implementation already includes a lot of casting and conversion logic.  It would be nice to tap into that and dispatch to the correct conversion code.  That's a project though.